### PR TITLE
Scavengers: prevent setting points to +/- Infinity

### PR DESCRIPTION
**Pull request by sparkychildcharlie**

### CODE REVIEW

File: chat-plugins/scavengers.js [hunk 1]
```diff
@@ -928,7 +928,7 @@ let commands = {
 		points = parseInt(points);
 
 		if (!targetId || targetId === 'constructor' || targetId.length > 18) return this.errorReply("Invalid username.");
-		if (!points || points < 0) return this.errorReply("Points must be an integer greater than 0.");
+		if (!points || points < 0 || points === Infinity) return this.errorReply("Points must be a non-Infinity integer greater than 0.");
```
Comment (Zarel) [inline chat-plugins/scavengers.js]: Infinity isn't an integer.

Comment (Zarel) [inline chat-plugins/scavengers.js]: Also "non-Infinity" is a silly way to say "finite".

Even so, "finite integer" is a silly way to say "integer".

Comment (Zarel) [inline chat-plugins/scavengers.js]: Also if you actually care about the value being a real integer, you might want to cap it at the point where doubles lose integer precision (2^53 i.e. 9,007,199,254,740,992)

Comment (xfix) [inline chat-plugins/scavengers.js]: It's possible to add two non-Infinity values and yet end up with Infinity. Wouldn't it be simpler to just put some cap on points possible to earn? Getting 34829521891282195912953269328683 points doesn't sound all that reasonable.

```
num = parseInt(1 + '7'.repeat(308))
1.7777777777777777e+308
num + num
Infinity
```

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**